src/layout/panel/workspaceList.js

```diff
@@ -553,26 +559,57 @@ var WorkspaceButton = GObject.registerClass(
         }
 
         handleDragOver(source, actor, x, y) {
-            if (!(source instanceof WorkspaceButton) || !this.draggable) {
-                return DND.DragMotionResult.NO_DROP;
+            if ((source instanceof WorkspaceButton) && this.draggable) {
+                this.emit('drag-over', y < this.height / 2);
+                return DND.DragMotionResult.MOVE_DROP;
             }
-
-            this.emit('drag-over', y < this.height / 2);
-            return DND.DragMotionResult.MOVE_DROP;
+            else if (source instanceof TaskBarItem) {
+                return DND.DragMotionResult.MOVE_DROP;
+            }
+            return DND.DragMotionResult.NO_DROP;
         }
 
         acceptDrop(source) {
-            if (!(source instanceof WorkspaceButton)) {
-                return false;
+            if (source instanceof WorkspaceButton) {
+                this.emit('drag-dropped');
+                return true;
             }
-            this.emit('drag-dropped');
-            return true;
+            else if (source instanceof TaskBarItem) {
+                const dragItem = dragData.current.item;
+                const tileableIndex = this.msWorkspace.tileableList.indexOf(dragItem.tileable);
+                const activeMsWorkspace = Me.msWorkspaceManager.getActiveMsWorkspace();
+                const isOnlyTileable = (activeMsWorkspace.tileableList.length == 2);
```

Comment:
Not sure this is a correct reason to discard the drop

Reply:
I tried many different procedures, this one is that works most of the times (in Wayland at least). If you have other suggestions, I can test it...

Reply:
My comment was about the prevent the drop in case the window was the last one. I mean now there is always an application launcher tileable which is undraggable so this check is unnecessary I believe.

But now I'm curious about your other issue

Reply:
What happen when it's don't work?

Reply:
Without this check you are able to drag the only tileable/window from a workspace to an empty one, what is unnecessary in my opinion.
You mean what happens when moving does not work? Same problem as opening some apps in Wayland (the app is not visible), please check issue #250.

Reply:
This check prevent from dragging the last window of the current workspace to anywhere no? For the bugs you mention we need to get more feedbacks since I don't reproduce this.

Reply:
I'll suggest to allow the dragging of the last window of the current workspace since I don't see a good reason to prevent that

Reply:
Even if there is only one workspace? This may prevent dropping by mistake as well...

Reply:
Well if you open a new window in the last workspace and you want to move it to another one in the first for exemple I don't see why you couldn't or if you have a single window opened in another monitor why you couldn't be able to move it to another workspace in the primary monitor?

Reply:
I didn't consider the multiple monitors situation. OK, then.

Reply:
Maybe I'm missing something but I feel that it's would be unnatural to prevent that